v2/common/src/test/java/com/google/cloud/teleport/v2/transforms/WriteToGCSTextTest.java

```diff
@@ -0,0 +1,118 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.transforms;
+
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.KV;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Test cases for the {@link WriteToGCSText} class. */
+@RunWith(JUnit4.class)
+public class WriteToGCSTextTest {
+
+  /** Rule for pipeline testing. */
+  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+
+  /** Rule for exception testing. */
+  @Rule public ExpectedException expectedException = ExpectedException.none();
+
+  private static final String FAKE_DIR = "FakeDirectory/";
+
+  private static final String TEXT_FILENAME = "TEXT-OUTPUT-";
+
+  private static final Integer NUM_SHARDS = 1;
+
+  private static final String FAKE_TEMP_LOCATION = "FakeTempLocation/";
+
+  // Create the test input.
+  private static final String key = "GenericRecord";
+  private static final String value = "Hey Bob";
+  private static final KV<String, String> message = KV.of(key, value);
+
+  /** Test whether {@link WriteToGCSText} throws an exception if no output directory is provided. */
+  @Test
+  public void testWriteWithoutOutputDirectory() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("withOutputDirectory(outputDirectory) called with null input.");
+
+    pipeline
+        .apply(
+            "CreateInput",
+            Create.of(message).withCoder(KvCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())))
+        .apply(
+            "WriteTextFile(s)",
+            WriteToGCSText.newBuilder()
+                .withOutputDirectory(null)
+                .withOutputFilenamePrefix(TEXT_FILENAME)
+                .withNumShards(NUM_SHARDS)
+                .withTempLocation(FAKE_TEMP_LOCATION)
+                .build());
+    pipeline.run();
+  }
+
+  /**
+   * Test whether {@link WriteToGCSText} throws an exception if number of shards are not provided.
+   */
+  @Test
+  public void testWriteWithoutNumShards() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("withNumShards(numShards) called with null input. ");
+
+    pipeline
+        .apply(
+            "CreateInput",
+            Create.of(message).withCoder(KvCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of())))
+        .apply(
+            "WriteTextFile(s)",
+            WriteToGCSText.newBuilder()
+                .withOutputDirectory(FAKE_DIR)
+                .withOutputFilenamePrefix(TEXT_FILENAME)
+                .withNumShards(null)
```

Comment:
@sabhyankar NumShards is a required parameter. So performing the required validation here by passing in Null.

Reply:
@sabhyankar Though of making NumShards a required parameter because when I was trying to set its default value to 0 it threw [java.lang.IllegalArgumentException: When applying WriteFiles to an unbounded PCollection, must specify number of output shards explicitly] this Exception.

Reply:
@sabhyankar 
I made the following changes 
1. Changed  withNumShards() to setNumShards().  
2. Setting its default value to be 0 in the autobuild.
3. Letting withNumShards() method from IO's (i.e. AvroIO, TextIO, FileIO) to handle this exception